pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +56,157 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
+	// because computed skew will be 3(3 - 0) if new Pod is scheduled to any of the three zones,
+	// it will violate MaxSkew.
+	//
+	// This is a beta field and requires the MinDomainsInPodTopologySpread feature gate to be enabled (enabled by default).
+	// +optional
+	MinDomains *int32 `json:"minDomains,omitempty" protobuf:"varint,5,opt,name=minDomains"`
+	// NodeAffinityPolicy indicates how we will treat Pod's nodeAffinity/nodeSelector
+	// when calculating pod topology spread skew. Options are:
+	// - Honor: only nodes matching nodeAffinity/nodeSelector are included in the calculations.
+	// - Ignore: nodeAffinity/nodeSelector are ignored. All nodes are included in the calculations.
+	//
+	// If this value is nil, the behavior is equivalent to the Honor policy.
+	// This is a beta-level feature default enabled by the NodeInclusionPolicyInPodTopologySpread feature flag.
+	// +optional
+	NodeAffinityPolicy *v1.NodeInclusionPolicy `json:"nodeAffinityPolicy,omitempty" protobuf:"bytes,6,opt,name=nodeAffinityPolicy"`
+	// NodeTaintsPolicy indicates how we will treat node taints when calculating
+	// pod topology spread skew. Options are:
+	// - Honor: nodes without taints, along with tainted nodes for which the incoming pod
+	// has a toleration, are included.
+	// - Ignore: node taints are ignored. All nodes are included.
+	//
+	// If this value is nil, the behavior is equivalent to the Ignore policy.
+	// This is a beta-level feature default enabled by the NodeInclusionPolicyInPodTopologySpread feature flag.
+	// +optional
+	NodeTaintsPolicy *v1.NodeInclusionPolicy `json:"nodeTaintsPolicy,omitempty" protobuf:"bytes,7,opt,name=nodeTaintsPolicy"`
+	// MatchLabelKeys is a set of pod label keys to select the pods over which
+	// spreading will be calculated. The keys are used to lookup values from the
+	// incoming pod labels, those key-value labels are ANDed with labelSelector
+	// to select the group of existing pods over which spreading will be calculated
+	// for the incoming pod. The same key is forbidden to exist in both MatchLabelKeys and LabelSelector.
+	// MatchLabelKeys cannot be set when LabelSelector isn't set.
+	// Keys that don't exist in the incoming pod labels will
+	// be ignored. A null or empty list means only match against labelSelector.
+	//
+	// This is a beta field and requires the MatchLabelKeysInPodTopologySpread feature gate to be enabled (enabled by default).
+	// +listType=atomic
+	// +optional
+	MatchLabelKeys []string `json:"matchLabelKeys,omitempty" protobuf:"bytes,8,opt,name=matchLabelKeys"`
+
+	//+optional
+	// +listType=atomic
+	// Defines what Prometheus Operator managed labels should be added to topologySpreadConstraints[].labelSelector.
+	AdditionalLabelSelectors []string `json:"additionalLabelSelectors,omitempty"`
+}
+
+func (tp TopologySpreadConstraint) BuildK8sTopologySpreadConstraint(podLabels map[string]string) v1.TopologySpreadConstraint {
+
+	if len(tp.AdditionalLabelSelectors) > 0 {
```

Comment:
[]string can be nil, so we also need to check for nil

Reply:
I've been looking on other part of codes and looks like it would only be nil if we mark it as a pointer, default array values are empty in go if I'm not wrong.

Reply:
Hmmm, I might be seeing things incorrectly, to be honest... I think empty slices when compared to nil will return true, but they won't return nilpointer exception when accessed?

Reply:
Looks like yes!
I wrote a unit test to ensure the behavior as you can see.
https://github.com/prometheus-operator/prometheus-operator/pull/5967/files#diff-e26bfd67af0e336825679ce12693ffeba2d292a7139d14bc20a18ea01af2fb05R212